modules/account-lib/test/unit/coin/dot/transaction.ts

```diff
@@ -54,6 +54,14 @@ describe('Dot Transaction', () => {
       tx.sender(DotResources.accounts.account2.address);
       should.deepEqual(tx.canSign({ key: DotResources.accounts.account2.secretKey }), true);
     });
+
+    it('can generate valid txHash from signed transaction', () => {
+      const signedTx = DotResources.rawTx.transfer.westendSigned2;
+      const txHash = Utils.default.getTxHash(signedTx);
```

Comment:
will it throw if it cannot generate the valid txHash? what is the expected behavior here

Reply:
It accepts any string and will return a txHash. We just have to trust that as long as we pass the same txHex we broadcast on chain then we will have a matching transaction id.